pkg/proto/replicator/doc.go

```diff
@@ -0,0 +1,2 @@
+// Package replicator contains the grpc protocol for interacting with bb_replicator
```

Comment:
What's the convention for documentation strings like these? Does it always have to start with `"Package %s"`, or would it be a bit more tidy to remove that prefix entirely?
```go
// The gRPC protocol for that BlobReplicator uses to communicate with instances of bb_replicator.
```

Nits:
- Please add a full stop to the end of every sentence.
- Make sure to wrap lines at <80 columns.
- s/grpc/gRPC/.

Reply:
> So I assume the package level documentation is what shows up in the Synopsis column on this page:
> 
> https://pkg.go.dev/github.com/buildbarn/bb-storage
> 
> Is there some automated way we can check that we don't forget to add these going forward? For example, is there some kind of golint option? By enabling that, we can ensure that we never forget to add these dummy files to the tree for Protobufs.

A little digging around indicates that the Synopsis column gets the first sentence of the package comment, and the whole comment shows up in the Overview section: https://pkg.go.dev/go.opentelemetry.io/otel@v0.11.0/api/correlation?tab=doc#pkg-overview

golint doesn't have any options, as part of its policy of being borderline unusable in CI (https://github.com/golang/lint#purpose), but https://staticcheck.io/docs/checks#ST1000 would cover it.

I am tempted to say: put golint in the bin, use https://golangci-lint.run/ and enable the interesting stuff, which also solves the tedious opinions it has about the bb_storage package name. However, that's probably a different PR, and I have a simpler solution in the short term: go mod tidy exits 1 on master because it can't find the empty packages. That's sufficient to test the thing we care about here, which is that go mod tidy can run to completion.

Reply:
> What's the convention for documentation strings like these? Does it always have to start with `"Package %s"`, or would it be a bit more tidy to remove that prefix entirely?

Interesting question, I had a look around to find out.

The standard library almost, but not quite, always uses this form: https://pkg.go.dev/std@go1.15.2?tab=packages

Other libraries are a bit more variable: https://pkg.go.dev/go.opentelemetry.io/otel@v0.11.0?tab=subdirectories

I don't have a strongly held opinion here, because realistically, these are probably placeholders which never get read by anybody. Protobuf packages aren't going to have their documentation in godoc, it would be produced by https://github.com/pseudomuto/protoc-gen-doc instead.

Reply:
> Nits:
> 
> * Please add a full stop to the end of every sentence.
> * Make sure to wrap lines at <80 columns.
> * s/grpc/gRPC/.

Duly tidied.

Reply:
These two files still need to be tidied up:

- pkg/proto/icas/doc.go
- pkg/proto/replicator/doc.go

Reply:
Got those two now.

golangci-lint is going to take more effort to beat into working with bazel, so I'll stick a pin in that idea for now.

> > What's the convention for documentation strings like these? Does it always have to start with `"Package %s"`, or would it be a bit more tidy to remove that prefix entirely?
> 
> Interesting question, I had a look around to find out.

It turns out that golint has strongly held opinions on this topic, but I'm rapidly losing interest in its opinions. I haven't puzzled out a workaround for it yet though.